site/content/blog/moving-existing-resources-to-a-new-app.md

```diff
@@ -0,0 +1,102 @@
+        Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app":"server", "kapp.k14s.io/app":"1645686658981326000"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable (reason: Invalid)
+```
+
+### Option 1
+To prevent this failure, add the [`kapp.k14s.io/disable-default-label-scoping-rules: ""`](/kapp/docs/latest/config/#labelscopingrules) annotation to individual resources (Deployments) to prevent kapp from touching the immutable fields when adopting them. Adding the annotation will exclude the resources from label scoping (used to scope resources within the current application).
+The annotation doesn't affect the app's ability to track such resources, the only consequence of disabling label scoping is that other resources in other apps cannot have the same label selector.
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  labels:
+    app: nginx
+  annotations:
+    kapp.k14s.io/disable-default-label-scoping-rules: ""
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.14.2
+        ports:
+        - containerPort: 80
+```
+
+### Option 2
+Another way to overcome this issue is to use `fallback-on-replace` as an [update strategy](/kapp/docs/latest/apply/#kappk14sioupdate-strategy), which will ask kapp to delete+create the resource on encountering an error while updating it. The update strategy can be used via adding the annotation `"kapp.k14s.io/update-strategy": "fallback-on-replace"`.
+The consequence of using this is having downtime as the deployment will be deleted and re-created.
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  labels:
+    app: nginx
+  annotations:
+    kapp.k14s.io/update-strategy: "fallback-on-replace"
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.14.2
+        ports:
+        - containerPort: 80
+```
+
+## Moving resources across kapp apps
+
+When we try to create an app with resources that are already part of another app, kapp will through an ownership error that says that the resource is already part of another app.
+To overcome this, you can use the flag `--dangerous-override-ownership-of-existing-resources` to override the ownership label on those resources. This would remove these resources from the other app and make them part of the current app.
+
+For some resources like Deployment, if label scoping was turned on (default=yes), then its spec.selector would also include app label, which would be tried to be rewritten when ownership changes. This will cause an immutable field error during update change since k8s deployment does not support changing selector labels.
```

Comment:
nit: spec.selector => `spec.selector`
(I think there is another mention of this YAML path above as well 🤔 )

Reply:
nit: "which would be tried to be rewritten when ownership changes" => "which would have to be  rewritten by `kapp`  when ownership changes"
(No strong opinions about this, up to you 🚀 )

Reply:
+1 to show a failure example and then how it can be rectified

Reply:
Done.